src/main/java/com/linecorp/armeria/client/metrics/MetricCollectingClient.java

```diff
@@ -0,0 +1,39 @@
+package com.linecorp.armeria.client.metrics;
+
+import java.util.function.Function;
+
+import com.codahale.metrics.MetricRegistry;
+
+import com.linecorp.armeria.client.Client;
+import com.linecorp.armeria.common.metrics.DropwizardMetricConsumer;
+
+
+public final class MetricCollectingClient {
```

Comment:
Let's extend `DecoratingClient` for now. At least it would make a String representation of a client look better. We could discuss this later.